six/modules/c++/cphd/include/cphd/Channel.h

```diff
@@ -23,72 +23,316 @@
+ */
+struct NoiseLevel
+{
+    NoiseLevel();
+
+    bool operator==(const NoiseLevel& other) const
+    {
+        return pnRef == other.pnRef &&
+                bnRef == other.bnRef &&
+                fxNoiseProfile == other.fxNoiseProfile;
+    }
+
+    bool operator!=(const NoiseLevel& other) const
+    {
+        return !((*this) == other);
+    }
+
+    double pnRef;
+    double bnRef;
+    mem::ScopedCopyablePtr<FxNoiseProfile> fxNoiseProfile;
+};
+
+/* Parameter Set that describes a CPHD data
+channel. Channels referenced by their unique
+Channel ID (Ch_ID). See Section 7.2. */
+struct ChannelParameter
+{
+    // Hiding TxRcv struct here because of name conflict //
+    struct TxRcv
+    {
+        bool operator==(const TxRcv& other) const
+        {
+            return txWFId == other.txWFId &&
+                    rcvId == other.rcvId;
+        }
+
+        bool operator!=(const TxRcv& other) const
+        {
+            return !((*this) == other);
+        }
+
+        std::vector<std::string> txWFId;
+        std::vector<std::string> rcvId;
+    };
+
+    // Hiding Antenna struct here because of naming clash //
+    struct Antenna
+    {
+        Antenna();
+
+        bool operator==(const Antenna& other) const
+        {
+            return txAPCId == other.txAPCId &&
+                    txAPATId == other.txAPATId &&
+                    rcvAPCId == other.rcvAPCId &&
+                    rcvAPATId == other.rcvAPATId;
+        }
+
+        bool operator!=(const Antenna& other) const
+        {
+            return !((*this) == other);
+        }
+
+        std::string txAPCId;
+        std::string txAPATId;
+        std::string rcvAPCId;
+        std::string rcvAPATId;
+    };
+
+    ChannelParameter();
+
+    bool operator==(const ChannelParameter& other) const
+    {
+        return identifier == other.identifier &&
+                refVectorIndex == other.refVectorIndex &&
+                fxFixed == other.fxFixed && toaFixed == other.toaFixed &&
+                srpFixed == other.srpFixed && signalNormal == other.signalNormal &&
+                polarization == other.polarization && fxC == other.fxC &&
+                fxBW == other.fxBW && fxBWNoise == other.fxBWNoise &&
+                toaSaved == other.toaSaved && toaExtended == other.toaExtended &&
+                dwellTimes == other.dwellTimes && imageArea == other.imageArea &&
+                antenna == other.antenna && txRcv == other.txRcv && tgtRefLevel == other.tgtRefLevel &&
+                noiseLevel == other.noiseLevel;
+    }
+
+    bool operator!=(const ChannelParameter& other) const
+    {
+        return !((*this) == other);
+    }
+
+    std::string identifier;
+    size_t refVectorIndex;
+    six::BooleanType fxFixed;
+    six::BooleanType toaFixed;
+    six::BooleanType srpFixed;
+    six::BooleanType signalNormal;
+    Polarization polarization;
+    double fxC;
+    double fxBW;
+    double fxBWNoise;
```

Comment:
Have some comment about `fxBWNoise` being optional

Reply:
Since these are structs with everything public, all the members need to have some sort of Doxygen comment. Nothing fancy, but see, e.g. six/sicd/RadarCollection.h for an example